app/bundles/LeadBundle/Model/TagModel.php

```diff
@@ -0,0 +1,135 @@
+<?php
+
+/*
+ * @copyright   2014 Mautic Contributors. All rights reserved
+ * @author      Mautic
+ *
+ * @link        http://mautic.org
+ *
+ * @license     GNU/GPLv3 http://www.gnu.org/licenses/gpl-3.0.html
+ */
+
+namespace Mautic\LeadBundle\Model;
+
+use Mautic\CoreBundle\Model\FormModel;
+use Mautic\LeadBundle\Entity\Tag;
+use Mautic\LeadBundle\Event\TagEvent;
+use Mautic\LeadBundle\Form\Type\TagEntityType;
+use Mautic\LeadBundle\LeadEvents;
+use Symfony\Component\EventDispatcher\Event;
+use Symfony\Component\HttpKernel\Exception\MethodNotAllowedHttpException;
+
+/**
+ * Class TagModel
+ * {@inheritdoc}
+ */
+class TagModel extends FormModel
+{
+    /**
+     * {@inheritdoc}
+     *
+     * @return string
+     */
+    public function getRepository()
+    {
+        return $this->em->getRepository(Tag::class);
+    }
+
+    /**
+     * {@inheritdoc}
+     *
+     * @return string
+     */
+    public function getPermissionBase()
+    {
+        return 'lead:leads';
+    }
+
+    /**
+     * Get a specific entity or generate a new one if id is empty.
+     *
+     * @param $id
+     *
+     * @return null|object
+     */
+    public function getEntity($id = null)
+    {
+        if (is_null($id)) {
```

Comment:
Should be a new method - You should know if you want saved entity or new one when calling this method. Could be problem when you want to throw 404 if tag was not found for example.

Reply:
That's convention for all Mautic models I'm afraid. See for example https://github.com/mautic/mautic/blob/staging/app/bundles/LeadBundle/Model/LeadModel.php#L394. I'm not sure if it's a good idea to break it in one model.